pandas/_libs/tslibs/conversion.pyx

```diff
@@ -290,10 +290,11 @@ cdef convert_to_tsobject(object ts, object tz, object unit,
         # Keep the converter same as PyDateTime's
         ts = datetime.combine(ts, time())
         return convert_datetime_to_tsobject(ts, tz)
-    elif is_period_object(ts):
-        raise ValueError("Cannot convert Period to Timestamp "
-                         "unambiguously. Use to_timestamp")
     else:
+        from .period import Period
```

Comment:
you can't use the is_period_object here, does tihs import make any perf difference?

Reply:
cimporting is_period_object at the module level raises ImportError at import-time.  The rules for when circular cimports are allowed are not clear to me.